lib/sdl_x11.hpp

```diff
@@ -69,35 +71,67 @@ class SdlX11Platform final : public SdlNativePlatform
       SDL_UnloadObject(lib);
 #endif
 #endif // SDL_VIDEO_DRIVER_X11_XINPUT2
+
+      // Create pipe for external events
+      if (pipe(event_pfd) == -1)
+      {
+         perror("pipe()");
+         exit(EXIT_FAILURE);
+      }
+      int flag = fcntl(event_pfd[0], F_GETFL);
+      fcntl(event_pfd[0], F_SETFL, flag | O_NONBLOCK);
    }
+
+   ~SdlX11Platform()
+   {
+      close(event_pfd[0]);
+      close(event_pfd[1]);
+   }
+
    void WaitEvent()
    {
-      int nstr, nfd = 1;
+      int nstr, nfd = 2;
       struct pollfd pfd[2];
 
       pfd[0].fd     = ConnectionNumber(disp);
       pfd[0].events = POLLIN;
       pfd[0].revents = 0;
-      if (glvis_command && visualize == 1)
-      {
-         pfd[1].fd     = glvis_command->ReadFD();
-         pfd[1].events = POLLIN;
-         pfd[1].revents = 0;
-         nfd = 2;
-      }
+
+      pfd[1].fd     = event_pfd[0];
+      pfd[1].events = POLLIN;
+      pfd[1].revents = 0;
+
       do
       {
          nstr = poll(pfd, nfd, -1);
```

Comment:
I found my answer in the man page but if others are interested on what happens here since `pdf[1]` is setup with `O_NONBLOCK` and `poll` is called with `timeout=-1` so it is blocking:

```
The poll() function shall not be affected by the O_NONBLOCK flag.
```